connectors/uplynk/src/main/java/com/theoplayer/android/connector/uplynk/internal/UplynkAdIntegration.kt

```diff
@@ -21,29 +24,44 @@ internal class UplynkAdIntegration(
     private val uplynkDescriptionConverter: UplynkSsaiDescriptionConverter,
     private val uplynkApi: UplynkApi
 ) : ServerSideAdIntegrationHandler {
+    private var pingScheduler: PingScheduler? = null
     private var adScheduler: UplynkAdScheduler? = null
     private val player: Player
         get() = theoplayerView.player
 
     init {
         player.addEventListener(PlayerEventTypes.TIMEUPDATE) {
-            adScheduler?.onTimeUpdate(it.currentTime.toDuration(DurationUnit.SECONDS))
+            val time = it.currentTime.toDuration(DurationUnit.SECONDS)
+            adScheduler?.onTimeUpdate(time)
+            pingScheduler?.onTimeUpdate(time)
+        }
+
+        player.addEventListener(PlayerEventTypes.SEEKING) {
+            pingScheduler?.onSeeking(it.currentTime.toDuration(DurationUnit.SECONDS))
+        }
+
+        player.addEventListener(PlayerEventTypes.SEEKED) {
+            pingScheduler?.onSeeked(it.currentTime.toDuration(DurationUnit.SECONDS))
         }
     }
 
     override suspend fun resetSource() {
         adScheduler = null
+        pingScheduler?.destroy()
     }
 
     override suspend fun setSource(source: SourceDescription): SourceDescription {
         adScheduler = null
+        pingScheduler?.destroy()
```

Comment:
Since we have the `resetSource()` implemented, isn't these 2 lines redundant?

Reply:
`resetSource()` is only called when we set the `source` to `null`. Not sure wether it was intended or it's a bug in the THEOplayer SSAI API.

This is why it's duplicated here

Reply:
Aha, didn't realize that.. I expect reset gets called whenever a new source is set regardless if it's a null or not.
Let's discuss internally for the THEOplayer SSAI API.